src/control/server/mgmt_pool_test.go

```diff
@@ -1129,23 +1129,136 @@ func TestServer_MgmtSvc_PoolQuery(t *testing.T) {
 	}
 }
 
-func TestServer_MgmtSvc_PoolSetProp(t *testing.T) {
-	withName := func(r *mgmtpb.PoolSetPropReq, n string) *mgmtpb.PoolSetPropReq {
-		r.SetPropertyName(n)
-		return r
-	}
-	withNumber := func(r *mgmtpb.PoolSetPropReq, n uint32) *mgmtpb.PoolSetPropReq {
-		r.SetPropertyNumber(n)
-		return r
-	}
-	withStrVal := func(r *mgmtpb.PoolSetPropReq, v string) *mgmtpb.PoolSetPropReq {
-		r.SetValueString(v)
-		return r
+func TestServer_MgmtSvc_PoolResolveID(t *testing.T) {
+	defaultLabel := "test-pool"
+
+	for name, tc := range map[string]struct {
+		req     *mgmtpb.PoolResolveIDReq
+		expResp *mgmtpb.PoolResolveIDResp
+		expErr  error
+	}{
+		"nil request": {
+			expErr: errors.New("nil request"),
+		},
+		"empty request": {
+			req:    &mgmtpb.PoolResolveIDReq{},
+			expErr: errors.New("empty request"),
+		},
+		"invalid label lookup": {
+			req: &mgmtpb.PoolResolveIDReq{
+				HumanID: "nope-bad-not-gonna-work",
+			},
+			expErr: errors.New("unable to find pool service"),
+		},
+		"valid label lookup": {
+			req: &mgmtpb.PoolResolveIDReq{
+				HumanID: defaultLabel,
+			},
+			expResp: &mgmtpb.PoolResolveIDResp{
+				Uuid: mockUUID,
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+
+			ms := newTestMgmtSvc(t, log)
+			addTestPools(t, ms.sysdb, mockUUID)
+			ps, err := ms.sysdb.FindPoolServiceByUUID(uuid.MustParse(mockUUID))
+			if err != nil {
+				t.Fatal(err)
+			}
+			ps.PoolLabel = defaultLabel
+			if err := ms.sysdb.UpdatePoolService(ps); err != nil {
+				t.Fatal(err)
+			}
+
+			gotResp, gotErr := ms.PoolResolveID(context.TODO(), tc.req)
+			common.CmpErr(t, tc.expErr, gotErr)
+			if tc.expErr != nil {
+				return
+			}
+
+			if diff := cmp.Diff(tc.expResp, gotResp, common.DefaultCmpOpts()...); diff != "" {
+				t.Fatalf("unexpected response (-want, +got)\n%s\n", diff)
+			}
+		})
 	}
-	withNumVal := func(r *mgmtpb.PoolSetPropReq, v uint64) *mgmtpb.PoolSetPropReq {
-		r.SetValueNumber(v)
-		return r
+}
+
+func propWithName(r *mgmtpb.PoolSetPropReq, n string) *mgmtpb.PoolSetPropReq {
+	r.SetPropertyName(n)
+	return r
+}
+func propWithNumber(r *mgmtpb.PoolSetPropReq, n uint32) *mgmtpb.PoolSetPropReq {
+	r.SetPropertyNumber(n)
+	return r
+}
+
+func propWithStrVal(r *mgmtpb.PoolSetPropReq, v string) *mgmtpb.PoolSetPropReq {
+	r.SetValueString(v)
+	return r
+}
+func propWithNumVal(r *mgmtpb.PoolSetPropReq, v uint64) *mgmtpb.PoolSetPropReq {
+	r.SetValueNumber(v)
+	return r
+}
+
+func TestServer_MgmtSvc_PoolSetProp_Label(t *testing.T) {
```

Comment:
name suggestion - How about adding "_Unique" onto the end, since that seems to be what this test is about?

Reply:
With the addition of idempotent label application logic, it's now not only about unique labels. The main goal here was to pull out testing of the label-specific scenarios to avoid making the regular property tests even more complicated.